apps/prairielearn/src/sync/fromDisk/assessments.ts

```diff
@@ -95,33 +96,37 @@ function getParamsForAssessment(
 
   let alternativeGroupNumber = 0;
   let assessmentQuestionNumber = 0;
-  const allRoleNames = (assessment.groupRoles ?? []).map((role) => role.name);
+  const allRoleNames = new Set((assessment.groupRoles ?? []).map((role) => role.name));
   const assessmentCanView = assessment?.canView ?? allRoleNames;
   const assessmentCanSubmit = assessment?.canSubmit ?? allRoleNames;
   const alternativeGroups = (assessment.zones ?? []).map((zone) => {
-    const zoneGradeRateMinutes = _.has(zone, 'gradeRateMinutes')
-      ? zone.gradeRateMinutes
-      : assessment.gradeRateMinutes || 0;
+    const zoneGradeRateMinutes =
+      Object.prototype.hasOwnProperty.call(zone, 'gradeRateMinutes') &&
+      zone.gradeRateMinutes !== undefined
+        ? zone.gradeRateMinutes
+        : assessment.gradeRateMinutes || 0;
     const zoneCanView = zone?.canView ?? assessmentCanView;
     const zoneCanSubmit = zone?.canSubmit ?? assessmentCanSubmit;
     return zone.questions.map((question) => {
       let alternatives: {
         qid: string;
-        maxPoints: number | number[];
-        points: number | number[];
-        maxAutoPoints: number | number[];
-        autoPoints: number | number[];
-        manualPoints: number;
+        maxPoints: number | null;
+        points: number | number[] | null;
+        maxAutoPoints: number | null;
+        autoPoints: number | number[] | null;
+        manualPoints: number | null;
         forceMaxPoints: boolean;
         triesPerVariant: number;
         gradeRateMinutes: number;
-        canView: string[] | null;
-        canSubmit: string[] | null;
+        canView: Set<string> | null;
+        canSubmit: Set<string> | null;
         advanceScorePerc: number;
       }[] = [];
-      const questionGradeRateMinutes = _.has(question, 'gradeRateMinutes')
-        ? question.gradeRateMinutes
-        : zoneGradeRateMinutes;
+      const questionGradeRateMinutes =
+        Object.prototype.hasOwnProperty.call(question, 'gradeRateMinutes') &&
```

Comment:
I think this check is probably unnecessary, `question.gradeRateMinutes !== undefined` should be sufficient? Ditto below. Heck, I think this could probably even be `question.gradeRateMinutes ?? zoneGradeRateMinutes`.

Reply:
Fixed in 8e81edefad398d73016f599105f0c449273a075f.

Reply:
Is there a reason you didn't go for the more concise nullish coalescing operator?

Reply:
Wouldn't that have differing behaviour? E.g. I have one set to `0` and a child property set to `1234`. In the original scenarios, it would be set to `0`, but if I use the nullish coalescing operator it would be `1234`?

Reply:
No, because `0` is not nullish. From a JS interpreter:

```
> 0 ?? 1234
0
```

Reply:
Ah, my apologies. I am thinking of the `||` operator. Thanks for the clarification.